core/common/src/main/java/alluxio/security/authentication/AuthenticatedUserInjector.java

```diff
@@ -77,6 +77,13 @@ public void onHalfClose() {
    * that is kept on auth-server.
    */
   private <ReqT, RespT> boolean authenticateCall(ServerCall<ReqT, RespT> call, Metadata headers) {
+    // Fail validation for closed/cancelled server calls.
+    if (!call.isReady() || call.isCancelled()) {
```

Comment:
Is checking `isReady()` appropriate here? That is just an indicator whether the server can send message immediately or not. It can be false when there are too many messages in the buffer.

Reply:
Nice catch. Just checking for cancelled now.